consensus/spos/bls/subroundEndRound.go

```diff
@@ -64,12 +80,24 @@ func checkNewSubroundEndRoundParams(
 	return err
 }
 
+func (sr *subroundEndRound) receivedHeader(headerHandler data.HeaderHandler) {
+	sr.AddReceivedHeader(headerHandler)
+
+	if !sr.IsSelfLeaderInCurrentRound() {
```

Comment:
I would add this check here: isConsensusHeaderReceived, header := sr.isConsensusHeaderReceived()

Reply:
I would let it there as the computation time of this method could be avoided by the previous checks which are done in the beginning of the doEndRoundJobByParticipant method